docs/conf.py

```diff
@@ -0,0 +1,120 @@
+# Configuration file for the Sphinx documentation builder.
+#
+# This file only contains a selection of the most common options. For a full
+# list see the documentation:
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+import os
+from datetime import date
+import sphinx_rtd_theme
+
+from sphinx.application import Sphinx
+from sphinx.transforms.post_transforms import SphinxPostTransform
+
+from mache.version import __version__
+
+# -- Project information -----------------------------------------------------
+
+project = "Mache"
+copyright = f"{date.today().year}, Energy Exascale Earth System Model Project"
+author = "E3SM Development Team"
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+if 'DOCS_VERSION' in os.environ:
+    version = os.environ.get('DOCS_VERSION')
+    release = version
+else:
+    # The short X.Y.Z version.
+    version = __version__
+    # The full version, including alpha/beta/rc tags.
+    release = __version__
+
+master_doc = "index"
+language = "en"
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    "myst_parser",
+    "sphinx_rtd_theme",
+    "sphinx_multiversion",
+    "sphinx.ext.autodoc",
+    "sphinx.ext.autosummary",
+    "sphinx.ext.intersphinx",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.mathjax",
+    "sphinx.ext.napoleon",
+]
+
+source_suffix = {
+    '.md': 'markdown',
+}
```

Comment:
Okay, this took *a while* to figure out.  This seems to mess up the API autogenerate docs.  They are ending up with a `.md` extension instead of the `.rst` extension they need to have.  I'd suggest removing this:
```suggestion
```
When I removed it (and fixed the spacing from 4 spaces to 3 in `api.md`), things finally are looking right for me.

Reply:
Will do. Is there somewhere I can see these issues you're having? I see them in the links that you've sent, but I don't know how to update that to see the changes I make. Everything looks fine when I build locally so I can't check there. Did you manually publish the docs somehow?

Reply:
I see! I see now that it wasn't correct on my end either, I just didn't even notice that the api documentation looked wrong. I switched these lines like you suggested and am now generating `.rst` files instead of `.md` files and it looks like you wanted.

I built with three and four spaces and I don't see a difference. From my searches online it looks like we *should* be able to use four spaces to indent as long as its consistent through the document. But I can't build on chrysalis right now so I can't check it anywhere except on my computer, so that might have an effect there. I'm just wondering if the issue was only with the `.md` line and not actually related to the spacing. The number of spaces is obviously not a big deal by any means, but if it's possible for it to be four spaces instead of three, it would be consistent through the rest of the documentation.

Reply:
Okay, great!

Let's stick with 4 spaces.  It gave me trouble in the past but maybe that was because the document sometimes used 3 and sometimes 4, or maybe it was a bug that's been fixed in the meantime.  Anyway, if it works for you, it will probably work fine on Chrysalis.  I changed it when trying to fix the problem and didn't bother changing it back when I found the actual cause of the problem.